### PR TITLE
Blazor server app example

### PR DESCRIPTION
**Pull request by zhiyuanliang-ms**
This app demonstrates how to use Feature Management in Blazor apps. It will display some enhanced contents for vip users.
image

Please read the included README.

### CODE REVIEW

File: examples/BlazorServerApp/Data/WeatherForecastService.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+namespace BlazorServerApp.Data
+{
+    public class WeatherForecastService
+    {
+        private static readonly string[] Summaries = new[]
+        {
+        "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
+    };
```
Comment (juniwang) [inline examples/BlazorServerApp/Data/WeatherForecastService.cs:8]: nit: indentation

File: src/Microsoft.FeatureManagement/FeatureManagementBuilder.cs [hunk 2]
```diff
@@ -37,15 +37,33 @@ public FeatureManagementBuilder(IServiceCollection services)
 
             if (!Services.Any(descriptor => descriptor.ServiceType == serviceType && descriptor.ImplementationType == implementationType))
             {
-                Services.AddSingleton(serviceType, implementationType);
+                //
+                // Register the feature filter with the same lifetime as the feature manager
+                if (!Services.Any(descriptor => descriptor.ServiceType == typeof(FeatureManager) && descriptor.Lifetime == ServiceLifetime.Scoped))
```
Comment (juniwang) [inline src/Microsoft.FeatureManagement/FeatureManagementBuilder.cs:42]: Are we registering `FeatureManager` as ServiceType or ImplementaionType? I'm expecting descriptor.ServiceType to be `typeof(IFeatureManager)`

File: src/Microsoft.FeatureManagement/FeatureManagementBuilder.cs [hunk 3]
```diff
@@ -37,15 +37,33 @@ public FeatureManagementBuilder(IServiceCollection services)
...
 
             return this;
         }
 
         public IFeatureManagementBuilder AddSessionManager<T>() where T : ISessionManager
         {
-            Services.AddSingleton(typeof(ISessionManager), typeof(T));
+            //
+            // Register the session manager with the same lifetime as the feature manager
+            if (!Services.Any(descriptor => descriptor.ServiceType == typeof(FeatureManager) && descriptor.Lifetime == ServiceLifetime.Scoped))
```
Comment (juniwang) [inline src/Microsoft.FeatureManagement/FeatureManagementBuilder.cs:59]: some doubt as above

File: src/Microsoft.FeatureManagement/FeatureManagementBuilder.cs (hunk 2 above)
Comment (zhiyuanliang-ms) [reply src/Microsoft.FeatureManagement/FeatureManagementBuilder.cs:42]: Yes, it should be IFeatureManager, Jimmy has called out this in this PR #303. I did not update it in this branch.

Comment (zhiyuanliang-ms) [reply src/Microsoft.FeatureManagement/FeatureManagementBuilder.cs]: Updated.

**Comment by zhiyuanliang-ms** [comment]
Plan to recreate an example project for Blazor

**Pull request closed without merging**